pkg/webhooks/machine_webhook.go

```diff
@@ -1027,6 +1029,30 @@ func validateAzureImage(image machinev1beta1.Image) []error {
 	return errors
 }
 
+func validateAzureDiagnostics(spec machinev1beta1.AzureDiagnostics, parentPath *field.Path) []error {
```

Comment:
nit: suggest to rename `spec` to smth like `diagnosticSpec`, just for make this name more descriptive within function.